awx/main/tasks.py

```diff
@@ -683,14 +683,7 @@ def build_inventory(self, instance, **kwargs):
                 json_data = json.dumps(instance.inventory.get_script_data(hostvars=True))
```

Comment:
@ryanpetrello a few lines above this, `if not os.path.exists(path):` now becomes highly suspicious. _If_ the path already existed, and it's the same content as a previous run, it could be subject to the fact caching bug.

I think this isn't a concern, because Ansible core has changed the default behavior anyway.

However, if `path` doesn't exist in the condition here, I would not expect it to work. That just wouldn't make any sense. Or possibly even worse, if you put a file named "inventory" in your playbook repo, this could trick us into using it here (but don't know, because I don't remember what each directory we use does).

Reply:
I'm trying to remember _why_ we did this - it was originally part of the isolated code.  Do you recall?

Reply:
> if you put a file named "inventory" in your playbook repo, this could trick us into using it here (but don't know, because I don't remember what each directory we use does)

This is a valid point.  We should probably change this inventory filename to be randomized, maybe something like:

```python
path = tempfile.mktemp(suffix='.inventory.py', dir=kwargs['private_data_dir'])
with open(path, 'w') as f:
   # always write the file
return path
```